java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -164,12 +167,21 @@ public static ArrowType getArrowTypeFromJdbcType(final JdbcFieldInfo fieldInfo,
       case Types.INTEGER:
         return new ArrowType.Int(32, true);
       case Types.BIGINT:
-        return new ArrowType.Int(64, true);
+        precision = fieldInfo.getPrecision();
+        if (precision > 18) {
+          return new ArrowType.Decimal(precision, 0);
```

Comment:
We ran into an issue with this on Snowflake. Although technically some 19 precision numbers can be expressed in 64 bit, the safest way to ensure the conversion always works seems to be doing this.

I can remove this, or add it in a separate PR if ya'll think think that's better.

Reply:
Nvm... I removed this since it broke some tests.